src/main/java/com/fasterxml/jackson/core/io/BigDecimalParser.java

```diff
@@ -40,7 +40,17 @@ private BigDecimalParser() {}
      * @throws NumberFormatException
      */
     public static BigDecimal parse(String valueStr) {
-        return parse(valueStr.toCharArray());
+        try {
+            if (valueStr.length() < 500) {
+                return new BigDecimal(valueStr);
+            }
+            return JavaBigDecimalParser.parseBigDecimal(valueStr);
```

Comment:
effectively, the same code as the parse(char[]) method but slight optimisation is that we avoid converting to a char array when we call the  JavaBigDecimalParser method